src/backend/distributed/commands/extension.c

```diff
@@ -115,3 +110,516 @@ ExtractNewExtensionVersion(Node *parsetree)
+											   &isNull);
+
+			if (!isNull)
+			{
+				text *versionText = DatumGetTextP(defaultVersionDatum);
+
+				return text_to_cstring(versionText);
+			}
+		}
+	}
+
+	/* couldn't find the name, this is not acceptable, Postgres would error as well */
+	ereport(ERROR,
+			(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+			 errmsg("invalid extension version name: \"%s\"", extname),
+			 errdetail("Version names must not be empty.")));
+
+	/* keep compilers happy */
+	return NULL;
+}
+
+
+/*
+ * FunctionCallGetTupleStore calls the given set-returning PGFunction and
+ * returns the ResultSetInfo filled by the called function.
+ */
+static ReturnSetInfo *
+FunctionCallGetTupleStore(PGFunction function, Oid functionId)
+{
+	LOCAL_FCINFO(fcinfo, 1);
+	FmgrInfo flinfo;
+	ReturnSetInfo *rsinfo = makeNode(ReturnSetInfo);
+	EState *estate = CreateExecutorState();
+	rsinfo->econtext = GetPerTupleExprContext(estate);
+	rsinfo->allowedModes = SFRM_Materialize;
+
+	fmgr_info(functionId, &flinfo);
+	InitFunctionCallInfoData(*fcinfo, &flinfo, 0, InvalidOid, NULL, (Node *) rsinfo);
+
+	(*function)(fcinfo);
+
+	return rsinfo;
+}
+
+
+/*
+ * Get the name of the schema that the coordinator would pick primarily
+ * for a CREATE EXTENSION statement that does not include "WITH SCHEMA"
+ * clause.
+ * Cannot return NULL, errors out as Postgres would do if NULL.
+ */
+static char *
+GetCurrentSchema(void)
+{
+	/*
+	 * Neither user nor author of the extension specified schema; use the
+	 * current default creation namespace, which is the first explicit
+	 * entry in the search_path.
+	 */
+	List *search_path = fetch_search_path(false);
+	Oid schemaOid = InvalidOid;
+	char *schemaName = NULL;
+
+	if (search_path == NIL) /* nothing valid in search_path? */
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_UNDEFINED_SCHEMA),
+				 errmsg("no schema has been selected to create in")));
+	}
+	schemaOid = linitial_oid(search_path);
+	schemaName = get_namespace_name(schemaOid);
+	if (schemaName == NULL) /* recently-deleted namespace? */
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_UNDEFINED_SCHEMA),
+				 errmsg("no schema has been selected to create in")));
+	}
+
+	list_free(search_path);
+
+	return pstrdup(schemaName);
+}
+
+
+/*
+ * ProcessCreateExtensionStmt is executed after the extension has been
+ * created locally and before we create it on the worker nodes.
+ * As we now have access to ObjectAddress of the extension that is just
+ * created, we can mark it as distributed to make sure that its
+ * dependencies exist on all nodes.
+ */
+void
+ProcessCreateExtensionStmt(CreateExtensionStmt *createExtensionStmt, const
+						   char *queryString)
+{
+	const ObjectAddress *extensionAddress = NULL;
+
+	if (!ShouldPropagateExtensionCommand((Node *) createExtensionStmt))
+	{
+		return;
+	}
+
+	extensionAddress = GetObjectAddressFromParseTree((Node *) createExtensionStmt, false);
+
+	EnsureDependenciesExistsOnAllNodes(extensionAddress);
```

Comment:
We should do something like but for extensions (please look at the source of the function below):
```C
EnsureSequentialModeForFunctionDDL();
```

Otherwise:
```SQL
CREATE EXTENSION isn ;
CREATE TABLE t1 (a int, b issn);
select create_distributed_table('t1', 'a');
...
DETAIL:  on server onderkalaci@localhost:9701
ERROR:  type "public.issn" does not exist
CONTEXT:  while executing command on localhost:9701
```

Reply:
Implemented EnsureSequentialModeForExtensionDDL  and utilized it in plan extension command functions in 359526463cf6a8c3559297b20839aba83800b2e7. Tests will be added soon

Reply:
tests for this are added in ae7bfdad2076bf2f55c150d9af7a830a13e8e18c but ineffective because of some other problems